### PR TITLE
Add delta_diff, file registry for delta files

### PR DESCRIPTION
**Pull request by bulv1ne**
### Description

Add `fileregistry::delta_diff` file registry. A different name might be better, this PR is a work in progress but 90% done.

I think I covered all test cases (no new data, new data, start date before first version).

### Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added **tests** that prove my fix is effective or that my feature works
- [x] I have added necessary **documentation** (if appropriate)
- [x] I have updated the **CHANGELOG.md**

### CODE REVIEW

File: getl/fileregistry/delta_diff.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+from ..block import BlockConfig
+from ..fileregistry import fileregistry_utils as fr_utils
+
+
+class DeltaDiff:
+    schema = T.StructType([T.StructField("version_date", T.TimestampType(), False)])
+
+    def __init__(self, bconf: BlockConfig) -> None:
+        self.file_registry_path = bconf.get("BasePath")
```
Comment (bulv1ne) [inline getl/fileregistry/delta_diff.py:14]: Could we deprecate BasePath in favour of FileRegistryPath or just Path? Both would work but BasePath gives a warning.

File: getl/fileregistry/fileregistry_utils.py [hunk 2]
```diff
@@ -16,8 +16,7 @@ def update_date_lifted(
 ) -> None:
     """Update date lifted column with the current date."""
     delta_table.delta_table.update(
-        F.col("date_lifted").isNull(),
-        {"date_lifted": "'{}'".format(str(datetime.now()))},
+        F.col("date_lifted").isNull(), {"date_lifted": F.lit(datetime.now())},
```
Comment (bulv1ne) [inline getl/fileregistry/fileregistry_utils.py:19]: I added this change as well, this shouldn’t be a problem right?

File: getl/fileregistry/delta_diff.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+from datetime import datetime
+
+from pyspark.sql import DataFrame, functions as F, types as T
+
+from ..block import BlockConfig
+from ..fileregistry import fileregistry_utils as fr_utils
```
Comment (ownbite) [inline getl/fileregistry/delta_diff.py]: This is a bit inconsistent with the rest of the repo where we use full paths.

File: getl/fileregistry/delta_diff.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        )
+        self.current_date = datetime.utcnow()
+
+        self.join_on_fields = bconf.get("JoinOnFields")
+
+        self.spark = bconf.spark
+
+        dataframe = fr_utils.fetch_file_registry(self.file_registry_path, self.spark)
+        if dataframe:
+            self.start_date = dataframe.limit(1).collect()[0].version_date
```
Comment (ownbite) [inline getl/fileregistry/delta_diff.py]: Will this always get the latest start_date? Should we add a failsafe of an order_by for example

**Review by ownbite — COMMENTED** [review]
I like the implementation, looks very nice.

File: getl/fileregistry/delta_diff.py (hunk 3 above)
Comment (bulv1ne) [reply getl/fileregistry/delta_diff.py]: :+1: will fix

File: getl/fileregistry/delta_diff.py (hunk 4 above)
Comment (bulv1ne) [reply getl/fileregistry/delta_diff.py]: Don't think a failsafe is needed, the whole delta table will be overwritten after every lift

**Pull request merged**